<a href="https://colab.research.google.com/github/IS2395/Python/blob/main/00.EJERCICIOS/EVALUACION4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 6 Práctica
Sentencias condicionales. (6 puntos)

## Ejercicio 1 (1.5 puntos)
Realizar un ejemplo de menú, donde podemos escoger las distintas opciones hasta que seleccionamos la opción de “Salir”.

Menú de recomendaciones

* Literatura
* Cine
* Música
* Videojuegos
* Salir

Al ingresar una opcion

opcion 1:

Lecturas recomendables:

* Esperándolo a Tito y otros cuentos de fútbol (Eduardo Sacheri)
* El juego de Ender (Orson Scott Card)
* El sueño de los héroes (Adolfo Bioy Casares)

opcion 2:

Películas recomendables:

* Matrix (1999)
* El último samuray (2003)
* Cars (2006)

opcion 3:

Discos recomendables:

* Despedazado por mil partes (La Renga, 1996)
* Búfalo (La Mississippi, 2008)
* Gaia (Mago de Oz, 2003)

opcion 4:

* Videojuegos clásicos recomendables
* Día del tentáculo (LucasArts, 1993)
* Terminal Velocity (Terminal Reality/3D Realms, 1995)
* Death Rally (Remedy/Apogee, 1996)

opcion 5:

Gracias, vuelva prontos

Opción no válida, en caso de ingresar un número fuera de las opciones



In [ ]:
#Tu codigo aqui
print('''MENU DE RECOMENDACIONES
1. Literatura
2. Cine
3. Musica
4. Videojuegos
5. Salir
Introduce el numero del menu que deseas visitar:  ''')
menu=int(input(''))
while menu != 5:
  if menu == 1:
    print('''Lecturas recomendables: 
    - Esperándolo a Tito y otros cuentos de fútbol (Eduardo Sacheri)
    - El juego de Ender (Orson Scott Card)
    - El sueño de los héroes (Adolfo Bioy Casares) ''')
    
  elif menu == 2:
    print('''Peliculas recomendables:
    - Matriz (1999)
    - El ultimo samurai (2003)
    - Cars (2006) ''')
  elif menu == 3:
    print('''Discos recomendables: 
    - Despedazado por mil partes (La Renga, 1996)
    - Búfalo (La Mississippi, 2008)
    - Gaia (Mago de Oz, 2003) ''')
  elif menu == 4:
    print('''Videojuegos clasicos recomendables: 
    - Día del tentáculo (LucasArts, 1993)
    - Terminal Velocity (Terminal Reality/3D Realms, 1995)
    - Death Rally (Remedy/Apogee, 1996)''')
  elif menu == 5:
    print('Gracias, vuelva pronto')
  else:
    print('Opcion no valida')
  print('''MENU DE RECOMENDACIONES
1. Literatura
2. Cine
3. Musica
4. Videojuegos
5. Salir
Introduce el numero del menu que deseas visitar:  ''')
  menu=int(input(''))
  

MENU DE RECOMENDACIONES
1. Literatura
2. Cine
3. Musica
4. Videojuegos
5. Salir
Introduce el numero del menu que deseas visitar:  
1
Lecturas recomendables: 
    - Esperándolo a Tito y otros cuentos de fútbol (Eduardo Sacheri)
    - El juego de Ender (Orson Scott Card)
    - El sueño de los héroes (Adolfo Bioy Casares) 
MENU DE RECOMENDACIONES
1. Literatura
2. Cine
3. Musica
4. Videojuegos
5. Salir
Introduce el numero del menu que deseas visitar:  
2
Peliculas recomendables:
    - Matriz (1999)
    - El ultimo samurai (2003)
    - Cars (2006) 
MENU DE RECOMENDACIONES
1. Literatura
2. Cine
3. Musica
4. Videojuegos
5. Salir
Introduce el numero del menu que deseas visitar:  
3
Discos recomendables: 
    - Despedazado por mil partes (La Renga, 1996)
    - Búfalo (La Mississippi, 2008)
    - Gaia (Mago de Oz, 2003) 
MENU DE RECOMENDACIONES
1. Literatura
2. Cine
3. Musica
4. Videojuegos
5. Salir
Introduce el numero del menu que deseas visitar:  
4
Videojuegos clasicos recomendables: 
    - D

## Ejercicio 2 (1.5 puntos)
Se pide por teclado un número y nos imprime los números primos que hay previos a este número.

Ejemplo: si ingresamos el 10 nos imprima del 1 a ese 10 cuales números son primos.


In [2]:
#Tu codigo aqui
numero=int(input('Ingresa un numero: '))
for i in range(2,numero+1):
  divisor=0
  for n in range(2,i):
    if i % n == 0:
      divisor=divisor+1
  if divisor == 0:
    print(f'El numero {i} es numero primo')
  else:
    print(f'El numero {i} no es primo')

Ingresa un numero: 10
El numero 2 es numero primo
El numero 3 es numero primo
El numero 4 no es primo
El numero 5 es numero primo
El numero 6 no es primo
El numero 7 es numero primo
El numero 8 no es primo
El numero 9 no es primo
El numero 10 no es primo



## Ejercicio 3 (1.5 puntos)
Una persona adquirió un producto para pagar en 20 meses. El primer mes pagó 10 €, el segundo 20 €, el tercero 40 € y así sucesivamente. Realizar un algoritmo para determinar cuánto debe pagar mensualmente y el total de lo que pagó después de los 20 meses.

Al finalizar los 20 meses pago en total:

primer mes pago 10, segundo mes pago 20, tercer mes pago 30, cuarto mes pago 40, etc.

In [ ]:
#Tu codigo aqui
pagos=[]
pago_mes=10
for mes in range (1,21):
  pagos.append(pago_mes)
  print (f'Pago en el mes {mes}: {pago_mes} €')
  pago_mes=pago_mes+10
total=sum(pagos)
print(f'El total que se pago fue de {total} €')

  




Pago en el mes 1: 10 €
Pago en el mes 2: 20 €
Pago en el mes 3: 30 €
Pago en el mes 4: 40 €
Pago en el mes 5: 50 €
Pago en el mes 6: 60 €
Pago en el mes 7: 70 €
Pago en el mes 8: 80 €
Pago en el mes 9: 90 €
Pago en el mes 10: 100 €
Pago en el mes 11: 110 €
Pago en el mes 12: 120 €
Pago en el mes 13: 130 €
Pago en el mes 14: 140 €
Pago en el mes 15: 150 €
Pago en el mes 16: 160 €
Pago en el mes 17: 170 €
Pago en el mes 18: 180 €
Pago en el mes 19: 190 €
Pago en el mes 20: 200 €
El total que se pago fue de 2100 €


#Ejercicio 4. (1.5 puntos) 

Crea una aplicación que pida un número y calcule su factorial (El factorial de un número es el producto de todos los enteros entre 1 y el propio número y se representa por el número seguido de un signo de exclamación. Por ejemplo 5! = 1x2x3x4x5=120).

In [ ]:
#Tu codigo aqui
numero=int(input('Ingresa en el numero del que quieres calcular su factorial: '))
resultado=1
for n in range(2,numero+1):
  resultado=resultado*n
print(f'{numero}!={resultado}')

Ingresa en el numero del que quieres calcular su factorial: 5
5!=120
